datajunction-server/datajunction_server/transpilation.py

```diff
@@ -0,0 +1,92 @@
+"""SQL transpilation plugins manager."""
+import importlib
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from datajunction_server.errors import DJException
+from datajunction_server.models.engine import Dialect
+
+
+class SQLTranspilationPlugin(ABC):
+    """
+    SQL transpilation plugin base class. To add support for a new SQL transpilation library,
+    implement this class with a custom `transpile_sql` method that works for the library. The
+    base implementation here will handle checking that the package exists before loading and
+    using it, or skipping transpilation if the package cannot be loaded.
+    """
+
+    package_name: Optional[str] = None
+
+    def __init__(self):
+        """
+        Load the package
+        """
+        self.check_package()
+
+    def check_package(self):
+        """
+        Check that the selected SQL transpilation package is installed and loads it.
+        """
+        if self.package_name:
+            try:
+                importlib.import_module(self.package_name)
+            except ImportError as import_err:
+                raise DJException(
+                    message=f"Not installed: {self.package_name}",
+                ) from import_err
+
+    @abstractmethod
+    def transpile_sql(
+        self,
+        query: str,
+        *,
+        input_dialect: Optional[Dialect] = Dialect.SPARK,
```

Comment:
So this default value shuld be `None`, correct?

Reply:
Good point, changed that!